src/v2/services/room/export-users.ts

```diff
@@ -0,0 +1,129 @@
+import { PhoneSMS } from "../../../constants/Config";
+import { FError } from "../../../error/ControllerError";
+import { ErrorCode } from "../../../ErrorCode";
+import { RoomModel } from "../../../model/room/Room";
+import { RoomUserModel } from "../../../model/room/RoomUser";
+import { UserPhoneModel } from "../../../model/user/Phone";
+import { UserModel } from "../../../model/user/User";
+import { EntityManager } from "typeorm";
+import { createLoggerService } from "../../../logger";
+import {
+    RoomExportUserItem,
+    RoomExportUsersReturn,
+    RoomInfoWithOwnerName,
+} from "./export-users.type";
+
+export class RoomExportUsersService {
+    private readonly logger = createLoggerService<"roomExportUsers">({
+        serviceName: "roomExportUsers",
+        ids: this.ids,
+    });
+
+    constructor(
+        private readonly ids: IDS,
+        private readonly DBTransaction: EntityManager,
+        private readonly userUUID: string,
+    ) {}
+
+    public async roomAndUsersIncludePhone(roomUUID: string): Promise<RoomExportUsersReturn> {
+        const room = await this.getRoomInfoIncludeOwnerName(roomUUID);
+        if (!room) {
+            this.logger.info("room not found", {
+                roomExportUsers: {
+                    userUUID: this.userUUID,
+                    roomUUID: roomUUID,
+                },
+            });
+            throw new FError(ErrorCode.RoomNotFound);
+        }
+
+        if (room.ownerUUID !== this.userUUID) {
+            this.logger.info("insufficient permissions,user is not room's owner", {
+                roomExportUsers: {
+                    userUUID: this.userUUID,
+                    roomUUID: roomUUID,
+                    roomOwnerUUID: room.ownerUUID,
+                },
+            });
+            throw new FError(ErrorCode.NotPermission);
+        }
+
+        const users = await this.getUsersByRoomUUID(roomUUID, this.phoneSMSEnabled);
+
+        return {
+            roomEndDate: room.roomEndDate,
+            roomStartDate: room.roomStartDate,
+            ownerName: room.ownerName,
+            roomTitle: room.roomTitle,
+            users: users,
+        };
+    }
+
+    private async getUsersByRoomUUID(
+        roomUUID: string,
+        includePhoneNumber: boolean,
+    ): Promise<RoomExportUserItem[]> {
+        const db = this.DBTransaction.createQueryBuilder(RoomUserModel, "ru")
+            .innerJoin(UserModel, "u", "ru.user_uuid = u.user_uuid")
+            .addSelect("u.user_name", "userName")
+            .addSelect("ru.created_at", "joinRoomDate");
+
+        if (includePhoneNumber) {
```

Comment:
```suggestion
    private async getUsersByRoomUUID(
        roomUUID: string,
    ): Promise<RoomExportUserItem[]> {
        const db = this.DBTransaction.createQueryBuilder(RoomUserModel, "ru")
            .innerJoin(UserModel, "u", "ru.user_uuid = u.user_uuid")
            .addSelect("u.user_name", "userName")
            .addSelect("ru.created_at", "joinRoomDate");

        if (PhoneSMS.enable) {
```

Reply:
I kept includePhoneNumber, which I thought would make it easier to write UT